tf2onnx/onnx_opset/math.py

```diff
@@ -54,62 +54,76 @@ def version_9(cls, ctx, node, **kwargs):
         pass
 
 
+def make_minmax_op(ctx, op_type, inputs, outputs,
+                   output_shapes=None, output_dtypes=None):
+    # support more dtype
+    supported_dtypes = [
+        onnx_pb.TensorProto.FLOAT,
+        onnx_pb.TensorProto.FLOAT16,
+        onnx_pb.TensorProto.DOUBLE
+    ]
+    target_dtype = onnx_pb.TensorProto.FLOAT
+    need_cast = False
+    cast_inputs = []
+    for inp in inputs:
+        dtype = ctx.get_dtype(inp)
+        utils.make_sure(dtype is not None, "dtype of {} is None".format(inp))
+        if dtype not in supported_dtypes:
+            cast_inp = ctx.make_node("Cast", [inp], attr={"to": target_dtype})
```

Comment:
will it be legal in tf while not in onnx that:  dtype of begin, end and stride is double, double, float? I have seen this is not supported in onnx slice.

Reply:
nope, it's also illegal in tf.